### PR TITLE
[v2.9] Add new agent TLS chart value & validation jsonSchema

### PR DESCRIPTION
**Pull request by mallardduck**
## Issue: 

 
TODO:
- Investigate how to handle "upgrades"
- Verify how the r/r internal code is going to handle this

### Automated Testing

* Test types added/modified:
    * Unit - Helm unit tests added to cover new value and possible results.

Summary: The tests verify all expected results based on potential valid inputs.

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: chart/templates/_helpers.tpl
Comment (MbolotSuse) [inline chart/templates/_helpers.tpl:106]: The translation here shouldn't be necessary - both the setting and the chart value should use strings and not booleans.

File: chart/templates/deployment.yaml
Comment (MbolotSuse) [inline chart/templates/deployment.yaml:144]: This should be `CATTLE_AGENT_TLS_MODE` and should be the same string value as was set in the chart.

File: chart/values.yaml
Comment (MbolotSuse) [inline chart/values.yaml:42]: I would leave the last item ("leave it undefined" off, and instead say "defaults to null, which does x").

File: chart/values.schema.json
Comment (MbolotSuse) [inline chart/values.schema.json:9]: I would avoid adding schema verification for this value. If you add something to this existing field you need to consider backward comparability and breaking changes, which I think is unnecessary/unhelpful for your use case.

**Review by MbolotSuse — CHANGES_REQUESTED** [review]
Overall comments:
- This is missing the logic to maintain `system-store` in case of upgrades
- The boolean translation is unnecessary - both the setting and the helm chart value should both be strings
-  Be careful on adding validation to "extraEnv" - since that's an existing value, you need to consider backward campatibility if you change that.

File: chart/values.schema.json
Comment (mallardduck) [reply chart/values.schema.json:9]: I thought you had mentioned us needing to add some sort of validation so people are less likely to incorrectly configure things. However that was before my unexpected time off and I was probably confused.  I'll remove this.

**Comment by mallardduck** [comment]
@MbolotSuse - if this looks good after re-review I will create PRs for 2.8/2.7 as well.

File: chart/tests/deployment_test.yaml
Comment (MbolotSuse) [inline chart/tests/deployment_test.yaml:482]: Nit: Might be good to add a test case for agentTLSMode: "".

**Review by MbolotSuse — COMMENTED** [review]
Small nits (give them a look but they don't need to be fixed before merge). Overall LGTM - I especially like the values schema, that was a good addition.

I will approve once https://github.com/rancher/rancher/pull/45684 is merged, since this depends on that PR.

**Comment by mallardduck** [comment]
@MbolotSuse tyvm. I've expanded the helm unit tests a bit more per suggestions.

File: chart/templates/_helpers.tpl
Comment (ericpromislow) [inline chart/templates/_helpers.tpl:76]: Please reinstate the final newline

**Review by ericpromislow — CHANGES_REQUESTED** [review]
I'd like a comment somewhere on how `chart/values.schema.json` is used. Searching doesn't find it, and because json doesn't allow comments, I guess it will have to go in a README.

File: chart/templates/_helpers.tpl
Comment (mallardduck) [reply chart/templates/_helpers.tpl]: fixed

**Comment by mallardduck** [comment]
> I'd like a comment somewhere on how `chart/values.schema.json` is used. Searching doesn't find it, and because json doesn't allow comments, I guess it will have to go in a README.

@ericpromislow  - Can you clarify what kind of documentation you would expect? This is a feature of helm [documented here](https://helm.sh/docs/topics/charts/#schema-files) and it uses JSON Schema which as a project has its own [documentation](https://json-schema.org/).

Also I'm not sure what readme it should live in since the main Rancher readme is intentionally minimalist. And this isn't a user-facing feature that they need to be aware of at the file level. Hopefully they will be aware of it as they run helm install commands and the output will make it more clear when they have an error with their values.

**Review by MbolotSuse — APPROVED** [review]

**Comment by ericpromislow** [comment]
ok, that clarified my question. No comment needed.

**Review by ericpromislow — APPROVED** [review]
LGTM with the understanding that the CI failure is not part of this PR

**Pull request merged**
